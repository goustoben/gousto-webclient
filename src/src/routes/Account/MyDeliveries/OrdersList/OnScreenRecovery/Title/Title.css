@value axiformaFontStack from 'styles/typography.css';
@value Black from 'styles/colors.css';

.title {
	font-family: axiformaFontStack;
	font-size: 20px;
	color: Black;
	line-height: 30px;
	font-weight: bold;
	text-align: center;
}
