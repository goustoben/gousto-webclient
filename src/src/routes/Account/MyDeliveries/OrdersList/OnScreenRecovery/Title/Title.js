import PropTypes from 'prop-types'
import React from 'react'

import css from './Title.css'

const propTypes = {
  title: PropTypes.string.isRequired,
}

const Title = ({ title }) => (
  <div className={css.title}>
    {title}
  </div>
)

Title.propTypes = propTypes

export { Title }
