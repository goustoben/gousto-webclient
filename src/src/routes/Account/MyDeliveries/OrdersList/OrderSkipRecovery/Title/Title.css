@value comfortaaFontStack from 'styles/typography.css';
@value Black from 'styles/colors.css';

.title {
	font-family: comfortaaFontStack;
	font-size: 20px;
	color: Black;
	line-height: 30px;
	font-weight: bold;
	text-align: center;
}
