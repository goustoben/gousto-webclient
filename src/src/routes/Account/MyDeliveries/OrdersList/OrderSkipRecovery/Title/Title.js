import React, { PropTypes } from 'react'

import { ModalTitle } from 'ModalComponent'

import css from './Title'

const prototype = {
    title: PropTypes.string.isRequired,
    orderType: PropTypes.string.isRequired,
}

const defaultProps = {
    title: '',
    orderType: '',
}

const Title = ({ title, orderType }) => {
    const titleCopy = title || `Are you sure you want to ${(orderType === 'pending') ? 'cancel' : 'skip'}?`

    return (
        <ModalTitle>
            <div className={css.title}>
                {titleCopy}
            </div>
        </ModalTitle>
    )
}

Title.prototype = prototype

Title.defaultProps = defaultProps

export default Title
