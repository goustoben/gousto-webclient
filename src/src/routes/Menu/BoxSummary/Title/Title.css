@value ScreenSMMin, ScreenMLMax, Screen3XSMax from 'styles/screenSizes.css';
@value sizeXS from 'styles/typography.css';

.desktopText {
	composes: noMargin from 'sass-loader!styles/spacing.scss';
	font-size: 15px;
}

.mobileText {
	composes: noMargin from 'sass-loader!styles/spacing.scss';
	color: White;
}

.title {
	composes: headerMD textUpper textBold from 'styles/typography.css';
}

.titledesktop {
	composes: title desktopText;
	color: inherit;
}

.titlemobile {
	composes: title mobileText;
	composes: textCenter from 'styles/typography.css';
	composes: paddingSMTop paddingSMLeftRight from 'sass-loader!styles/spacing.scss';
	composes: relative from 'styles/position.css';
	z-index: 1000;
}
.titleWrapper {
	composes: flex verticalCenter from 'styles/flex.css';
}

.showDate {
	visibility: visible;
	min-width: 85px;
	@media Screen3XSMax {
		font-size: sizeXS;
	}
}

.hideDate {
	visibility: hidden;
}

.price {
	composes: paddingSMLeft from 'sass-loader!styles/spacing.scss';
	margin-left: auto;
	text-align: right;
	display: inline-block;
	min-width: 120px;
	@media ScreenSMMin and ScreenMLMax {
		min-width: auto
	}
}

.spinnerContainer {
	display: none;
	padding: 0;
	position: absolute;
	width: 100%;
	height: 100%;
	opacity: 0;
	bottom: 0;
	top: 0;
	border-color: transparent;
}

.spinner {
	display: block;
	width: 36px;
	height: 100%;
	position: relative;
	transform: translateY(-50%);
	margin-left: auto;
	margin-right: 24%;
}

.spinnerShow {
	opacity: 0.8;
	z-index: 99;
	display: block;
}
