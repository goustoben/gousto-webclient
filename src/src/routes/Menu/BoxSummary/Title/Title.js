import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'
import moment from 'moment'

import { Spinner } from 'goustouicomponents'
import { isMobile } from 'utils/view'
import Price from '../Price'
import css from './Title.css'

class Title extends React.PureComponent {
  static propTypes = {
    date: PropTypes.string,
    view: PropTypes.string,
    spinnerClassName: PropTypes.string,
    spinnerContainerClassName: PropTypes.string,
    pending: PropTypes.bool,
    recipeTotal: PropTypes.number,
    recipeDiscount: PropTypes.number,
    recipeTotalDiscounted: PropTypes.number,
    slotTime: PropTypes.string,
  }
  static defaultProps = {
    date: '',
    view: '',
    spinnerClassName: '',
    spinnerContainerClassName: '',
    slotTime: '',
    pending: false,
    recipeTotal: 0,
    recipeDiscount: 0,
    recipeTotalDiscounted: 0,
  }

  render() {
    const { recipeTotal, recipeDiscount, recipeTotalDiscounted, slotTime } = this.props
    const date = this.props.date
    const view = this.props.view
    const pending = this.props.pending
    const spinnerClassNames = {
      [css.spinnerContainer]: true,
      [this.props.spinnerContainerClassName]: true,
      [css.spinnerShow]: pending,
    }

    return (
      <div className={css.titleWrapper}>
        <p className={css[`title${view}`]}>
          {date ? <span className={css.showDate}>{moment(date).format('ddd D MMM')}</span> : null}
          {slotTime ? <span className={css.showDate}> {slotTime}</span> : null}
        </p>
        <div className={classNames(css[`title${view}`], css.price)}>
          {
            pending ? isMobile(view) && (
              <div className={classNames(spinnerClassNames)}>
                <span className={classNames(css.spinner, this.props.spinnerClassName)}>
                  <Spinner />
                </span>
              </div>
            ) : <Price
              recipeTotal={recipeTotal}
              recipeDiscount={recipeDiscount}
              recipeTotalDiscounted={recipeTotalDiscounted}
            />
          }
        </div>
      </div>
    )
  }

}
export default Title
