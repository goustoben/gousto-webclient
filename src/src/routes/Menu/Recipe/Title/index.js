export { Title } from './Title'
export { TitleContainer } from './TitleContainer'
