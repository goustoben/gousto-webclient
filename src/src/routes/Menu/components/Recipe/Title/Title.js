import React from 'react'
import PropTypes from 'prop-types'
import { useRecipeField } from 'routes/Menu/context/recipeContext'

const Title = ({ className }) => {
  const title = useRecipeField('title')

  return <h2 className={className}>{title}</h2>
}

Title.propTypes = {
  className: PropTypes.string
}

Title.defaultProps = {
  className: undefined
}

export { Title }
