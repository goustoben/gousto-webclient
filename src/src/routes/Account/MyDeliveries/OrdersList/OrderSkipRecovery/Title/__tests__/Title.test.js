import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Title from '../Title'

describe('Order Skip Recovery Model Title', () => {
  let wrapper

  describe('Innitial Render', () => {
    test('should render snapshot', () => {
      const tree = renderer.create(
				<Title />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    test('should render title', () => {
      wrapper = mount(<Title />)

      expect(wrapper.text()).toBe('Are you sure you want to skip?')
    })
  })

  describe('Alternative Render', () => {
    test('should render title to say "cancel" instead of "skip"', () => {
      wrapper = mount(<Title orderType="pending" />)

      expect(wrapper.text()).toBe('Are you sure you want to cancel?')
    })

    test('should render custom title', () => {
      wrapper = mount(<Title title="custom title" />)

      expect(wrapper.text()).toBe('custom title')
    })

    test('should prioritise title over orderType', () => {
      wrapper = mount(<Title title="foo bar" orderType="pending" />)

      expect(wrapper.text()).toBe('foo bar')
    })
  })
})
