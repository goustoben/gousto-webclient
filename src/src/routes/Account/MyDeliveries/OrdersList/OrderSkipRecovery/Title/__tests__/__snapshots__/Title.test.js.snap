// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Skip Recovery Model Title Innitial Render should render snapshot 1`] = `
<div
  className="modalTitle"
>
  <div>
    Are you sure you want to skip?
  </div>
</div>
`;
